### PR TITLE
Test building external plugins in CI

### PR DESCRIPTION
**Pull request by dominiklohmann**
###  :notebook_with_decorative_cover: Description

Before this change, we only tested building shared and static plugin libraries alongside VAST in CI. This sets up a CI workflow that downloads a previously created VAST artifact, and builds and tests plugins against that. For now, this is only done for the example plugin.

###  :memo: Checklist

- [x] All user-facing changes have changelog entries.
- [x] The changes are reflected on [docs.tenzir.com/vast](https://docs.tenzir.com/vast), if necessary.
- [x] The PR description contains instructions for the reviewer, if necessary.

### :dart: Review Instructions

File-by-file.

### CODE REVIEW

File: .github/workflows/vast.yaml [hunk 1]
```diff
@@ -1,4 +1,4 @@
-name: "VAST"
+name: 'VAST'
```
Comment (0snap) [inline .github/workflows/vast.yaml:1]: There's no need to quote strings behind the colon.

File: .github/workflows/vast.yaml [hunk 2]
```diff
@@ -513,3 +522,94 @@ jobs:
           # for builds of the latest release.
           asset_name: "${{ steps.configure_env.outputs.publish_name }}.tar.gz"
           asset_content_type: application/gzip
+
+  plugins:
+    runs-on: ubuntu-20.04
+    # run everything inside a Debian container on an ubuntu-20.04 host
```
Comment (0snap) [inline .github/workflows/vast.yaml:528]: this comment should be moved down by one line.

File: .github/workflows/vast.yaml [hunk 3]
```diff
@@ -513,3 +522,94 @@ jobs:
...
+          cmake --build "$BUILD_DIR" --target all --parallel
+      - name: Run Unit Tests
+        run: |
+          cmake --build "$BUILD_DIR" --target test
+      - name: Run Integration Tests
+        id: integration_tests
+        run: |
+          echo "::set-output name=running=true"
+          cmake --build "$BUILD_DIR" --target integration
+          echo "::set-output name=running=false"
```
Comment (0snap) [inline .github/workflows/vast.yaml:605]: What do we need the `running` output for?

File: .github/workflows/vast.yaml [hunk 4]
```diff
@@ -513,3 +522,94 @@ jobs:
...
+      - name: Upload Integration Test Logs on Failure
+        if: failure()
+        uses: actions/upload-artifact@v2
+        with:
+          name: 'vast-${{ matrix.plugin.target }}-integration-test'
+          path: '${{ env.BUILD_DIR }}/vast-${{ matrix.plugin.target }}-integration-test'
+          if-no-files-found: ignore
+      - name: Install
+        run: |
+          cmake --install "$BUILD_DIR" --prefix "$INSTALL_DIR"
```
Comment (0snap) [inline .github/workflows/vast.yaml:615]: Why don't we upload the build artifacts to GitHub at the end? The example plugin binary artifact might be a good source to get folks started with running VAST with a plugin (we could add a docs section).

**Review by 0snap — CHANGES_REQUESTED** [review]
Looking good, nothing major from my side. I left some questions in the file.

File: .github/workflows/vast.yaml (hunk 4 above)
Comment (dominiklohmann) [reply .github/workflows/vast.yaml:615]: They are not very portable and other than VAST the libraries are not meant to be built against. I intentionally decided against that.

File: .github/workflows/vast.yaml (hunk 3 above)
Comment (dominiklohmann) [reply .github/workflows/vast.yaml:605]: Leftover from a previous test, will remove.

File: .github/workflows/vast.yaml (hunk 2 above)
Comment (dominiklohmann) [reply .github/workflows/vast.yaml:528]: I opted to remove it, it's a pretty meaningless comment.

**Review by 0snap — APPROVED** [review]
LGTM :rocket:

**Pull request merged**